format/index.md

```diff
@@ -52,761 +58,507 @@ called out below.
-Additional filtering allowed by an API should be specified in its profile (see
-[Extending](/extending)).
+If a client requests a restricted set of fields, an endpoint **MUST NOT** include other
+fields in the response.
 
-### Inclusion of Linked Resources <a href="#fetching-includes" id="fetching-includes" class="headerlink"></a>
-
-A server **MAY** choose to support returning compound documents that include
-both primary and linked resource objects.
-
-An endpoint **MAY** return resources linked to the primary resource(s) by
-default.
-
-An endpoint **MAY** also support custom inclusion of linked resources based upon
-an `include` request parameter. This parameter should specify the path to one or
-more resources relative to the primary resource. If this parameter is used,
-**ONLY** the requested linked resources should be returned alongside the primary
-resource(s).
-
-For instance, comments could be requested with a post:
+### Sorting <a href="#fetching-sorting" id="fetching-sorting" class="headerlink"></a>
 
-```text
-GET /posts/1?include=comments
-```
+A server **MAY** choose to support requests to sort resource collections
+according to one or more criteria.
 
-In order to request resources linked to other resources, the dot-separated path
-of each relationship should be specified:
+An endpoint **MAY** support requests to sort the primary data with a `sort`
+query parameter.
 
 ```text
-GET /posts/1?include=comments.author
+GET /people?sort=+age
 ```
 
-Note: a request for `comments.author` should not automatically also include
-`comments` in the response (although comments will obviously need to be queried
-in order to fulfill the request for their authors).
-
-Multiple linked resources could be requested in a comma-separated list:
+An endpoint **MAY** support multiple sort criteria by allowing
+comma-separated (U+002C COMMA, ",") fields as the value for `sort`. Sort
+criteria should be applied in the order specified.
 
 ```text
-GET /posts/1?include=author,comments,comments.author
+GET /people?sort=+age,+name
 ```
 
-### Sparse Fieldsets <a href="#fetching-sparse-fieldsets" id="fetching-sparse-fieldsets" class="headerlink"></a>
-
-A server **MAY** choose to support requests to return only specific fields in
-resource object.
-
-An endpoint **MAY** support requests that specify fields for the primary
-resource type with a `fields` parameter.
+The sort order for each field **MUST** be specified with one of the
+following prefixes:
 
-```text
-GET /people?fields=id,name,age
-```
+* Plus (U+002B PLUS SIGN, "+") to request an ascending sort order.
+* Minus (U+002D HYPHEN-MINUS, "-") to request a descending sort order.
 
-An endpoint **MAY** support requests that specify fields for any resource type
-with a `fields[TYPE]` parameter.
+> Note: By requiring a sort order prefix instead of allowing a default
+order, JSON API avoids setting requirements for the first character in field
+names. 
```

Comment:
It's not clear to me why this is a good trade-off. I don't see that it's particularly bad, either, but does put the added burden on the client to explicitly state a sort order every time.

If JSON API were to set requirements for the first character in field names, it seems like they would tend to match existing programming language requirements. Requiring that field names not start with PLUS, MINUS, or other non-numerical characters doesn't seem like a huge overreach.